python/dlr/dlr_model.py

```diff
@@ -87,16 +70,47 @@ def _lazy_init_output_shape(self):
 
     def _parse_backend(self):
         backend = c_char_p()
-        _check_call(_LIB.GetDLRBackend(byref(self.handle),
+        self._check_call(self._lib.GetDLRBackend(byref(self.handle),
                                        byref(backend)))
         return backend.value.decode('ascii')
 
     def _get_version(self):
         version = c_char_p()
-        _check_call(_LIB.GetDLRVersion(byref(version)))
+        self._check_call(self._lib.GetDLRVersion(byref(version)))
         return version.value.decode('ascii')
 
-    def __init__(self, model_path, dev_type='cpu', dev_id=0):
+    def _check_call(self, ret):
+        """
+        Check the return value of C API call
+        This function will raise exception when error occurs.
+        Wrap every API call with this function
+
+        Parameters
+        ----------
+        ret : int
+            return value from API calls
+        """
+        if ret != 0:
+            raise DLRError(self._lib.DLRGetLastError().decode('ascii'))
+
+    def _init_libdlr(self):
+        paths = find_lib_path()
```

Comment:
Can we move it to line 109?

Reply:
Could you elaborate?

Reply:
calling `find_lib_path()` is needed only If `not self.use_default_dlr and os.path.exists(local_libdlr_so_path)` is false. We can move `find_lib_path()` to `else` block - line 109